src/figuresClasses.ts

```diff
@@ -1,11 +1,77 @@
-export interface Figure {}
+export interface Figure {
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: string = 'triangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('your error message');
+    }
+
+    if (a + b <= c || a + c <= b || c + b <= a) {
+      throw new Error('your error message');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const result = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.round(result * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  public shape: string = 'circle';
 
-export class Triangle implements Figure {}
+  constructor(
+    public color: string,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('your error message');
+    }
+  }
 
-export class Circle implements Figure {}
+  getArea(): number {
+    const result = Math.PI * this.radius ** 2;
+
+    return Math.floor(result * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  public shape: string = 'rectangle';
+
+  constructor(
+    public color: string,
+    public width: number,
+    public height: number,
+  ) {
```

Comment:
[CODE KNOWLEDGE] - TypeScript automatically creates properties from constructor parameters with access modifiers. Manual assignment is redundant.